### ISSUE TITLE
privacyidea-usercache-cleanup @ python3.12 : AttributeError: module 'inspect' has no attribute 'getargspec'. Did you mean: 'getargs'?

### ISSUE THREAD

**Issue by beckerr-rzht** [privacyidea/privacyidea#4015]
### Top-level intent

Running `privacyidea-usercache-cleanup`

### Steps to reproduce

```bash
privacyidea-usercache-cleanup
```

### Actual outcome

```bash
$ privacyidea-usercache-cleanup
Can not import grpc modules.
Traceback (most recent call last):
  File "/opt/privacyidea/bin/privacyidea-usercache-cleanup", line 60, in <module>
    @manager.command
     ^^^^^^^^^^^^^^^
  File "/opt/privacyidea/lib/python3.12/site-packages/flask_script/__init__.py", line 288, in command
    command = Command(func)
              ^^^^^^^^^^^^^
  File "/opt/privacyidea/lib/python3.12/site-packages/flask_script/commands.py", line 118, in __init__
    args, varargs, keywords, defaults = inspect.getargspec(func)
                                        ^^^^^^^^^^^^^^^^^^
AttributeError: module 'inspect' has no attribute 'getargspec'. Did you mean: 'getargs'?
```

I think that `inspect.getfullargspec` would be the right choice.
https://docs.python.org/3/library/inspect.html#inspect.getfullargspec

### Configuration

* **privacyIDEA version**: 3.9.3
* **Installation method**: venv pip
* **Python version**: 3.12
* **Operating system**: Ubuntu 24.04 LTS
* **Webserver**: Apache2
* **Token database**: MariaDB 10.11 + Galera 26.4.18

**Comment by plettich** [comment]
Yes, this is a known issue with privacyIDEA 3.9 and Python > 3.10
We used some external package which isn't compatible with Python > 3.10. We have replaced this package for privacyIDEA 3.10 which is due to be released soon (#3593 ).
Until then, don't run privacyIDEA with Python > 3.10 or patch the external package to use `getfullargspec()`.

**Comment by beckerr-rzht** [comment]
All right.

Just in case anyone needs it, here's a suitable patch:
```diff
--- lib/python3.12/site-packages/flask_script/commands.py.orig  2024-07-24 21:21:50.378297287 +0200
+++ lib/python3.12/site-packages/flask_script/commands.py       2024-07-24 20:58:56.388639117 +0200
@@ -115,7 +115,7 @@
                 self.option_list = []
             return
 
-        args, varargs, keywords, defaults = inspect.getargspec(func)
+        args, varargs, keywords, defaults, _, _, _ = inspect.getfullargspec(func)
         if inspect.ismethod(func):
             args = args[1:]
 
```
